src/main/java/Duke.java

```diff
@@ -1,10 +1,313 @@
+        for (String s : statement) {
+            printStatement(s);
+        }
+        printHorizontalLine();
+    }
+
+    private static void printError(String errorDescription) {
+        printHorizontalLine();
+        printStatement(String.format("%s %s", ERROR_PREFIX_MESSAGE, errorDescription));
+        printHorizontalLine();
+    }
+
+    private static void printWelcomeMessage() {
+        printStatements(LOGO, WELCOME_MESSAGE);
+    }
+
+    private static void exitProgram() {
+        printStatements(EXIT_MESSAGE);
+        System.exit(0);
+    }
+
+    /**
+     * Parses the user input and attempts to execute the command
+     * with the arguments.
+     */
+    private static void executeCommand(String userInput) {
+        String[] parsedCommand = parseCommand(userInput);
+        String commandName = parsedCommand[0];
+        String commandArgs = parsedCommand[1];
+
+        switch (commandName) {
+        case COMMAND_LIST_WORD:
+            listTasks();
+            break;
+        case COMMAND_TODO_WORD:
+            recordTodo(commandArgs);
+            break;
+        case COMMAND_DEADLINE_WORD:
+            recordDeadline(commandArgs);
+            break;
+        case COMMAND_EVENT_WORD:
+            recordEvent(commandArgs);
+            break;
+        case COMMAND_DONE_WORD:
+            markTaskDone(commandArgs);
+            break;
+        case COMMAND_BYE_WORD:
+            exitProgram();
+        default:
+            printError(ERROR_COMMAND_MESSAGE);
+        }
+    }
+
+    /**
+     * Splits the user input into a command word and command arguments string.
+     *
+     * @return an array of size 2; first element is the command word
+     *         and second element is the arguments string
+     */
+    private static String[] parseCommand(String userInput) {
+        final String[] parsedCommand = userInput.trim().split("\\s+", 2);
+        // parsedCommand's length after split will always be more than zero.
+        if (parsedCommand.length == 1) {
+            return new String[] {parsedCommand[0], ""};
+        }
+        return parsedCommand;
+    }
+
+    /**
+     * Parses an argument value after a given token.
+     * Reads until a next token or the end of commandArgs string.
+     * If token is null, read from the start of the string until
+     * a next token or the end of commandArgs string.
+     *
+     * @param commandArgs a full string of command arguments
+     * @param token a string representing an option portion of an argument e.g. "/by"
+     * @return the argument value after the given token
+     */
+    private static String parseArgument(String commandArgs, String token) {
+        int readFromIndex = 0;
+        int readUntilIndex = commandArgs.length();
+        if (token != null) {
+            if (commandArgs.contains(token)) {
+                readFromIndex = commandArgs.indexOf(token) + token.length();
+            } else {
+                return null;
+            }
+        }
```

Comment:
I think it is better to separate the conditional statements into multiple individual ones and make the main code of assigning the readFromIndex value more prominent for easier management and readability.

Reply:
Thanks for the advice!